.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,94 @@
+name: Pipeline deployment
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches: [master]
+    types: [opened, synchronize]
+
+env:
+  ML_IMAGE_TAG: activelearning_ml:test
+  API_IMAGE_TAG: activelearning_api:test
+  ML_ENDPOINT: train_and_predict/
+  API: api/
+  WEBAPP: webapp/
+
+jobs:
+  linting:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8' 
+          architecture: 'x64'
+          cache: 'pip'
+      - name: flake8 Lint
+        uses: py-actions/flake8@v2
```

Comment:
Thanks for sharing this tool, it's really helpful! So to confirm, I'll require a setup.cfg and yml file in the root directory folder? Also, the linting for the webapp using cypress is currently directly included in the action. Should it be integrated into the pre-commit ci tool that you have suggested, or is it ok to leave it there for now? I was thinking that it can be changed in another PR.

Reply:
> I'll require a setup.cfg and yml file in the root directory folder?

Yes, `setup.cfg` with flake8 config and `.yml` with pre-commit config.



> Should I make the change in this PR?

Let's extract linting into separate PR, we might merge it faster than everything else.



> Should it be integrated into the pre-commit ci tool that you have suggested, or is it ok to leave it there for now? I was thinking that it can be changed in another PR.

Let's keep it as it is for now, it is already working. But if you can, add it to the pre-commit as well later. I haven't used eslint before but a quick search suggests it [should be easy](https://github.com/pre-commit/mirrors-eslint). (Or maybe if you think its's easy you can do it in the linting PR)